### PR TITLE
`gardener-scheduler` provides meaningful error message when no seeds with matching providers found

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area ops-productivity
/kind enhancement

**What this PR does / why we need it**:
`gardener-scheduler` now provides a meaningful error message when no seeds with matching providers found.

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/scheduler/controller/shoot/reconciler.go [hunk 1]
```diff
@@ -225,6 +225,10 @@ func filterSeedsMatchingProviders(cloudProfile *gardencorev1beta1.CloudProfile,
 			matchingSeeds = append(matchingSeeds, seed)
 		}
 	}
+
+	if len(matchingSeeds) == 0 {
+		return nil, fmt.Errorf("none out of the %d seeds has a matching provider for %q", len(seedList), shoot.Spec.Provider.Type)
```
Comment (acumino) [inline pkg/scheduler/controller/shoot/reconciler.go:230]: I don't think we need to show no of seeds in the error message.
```suggestion
		return nil, fmt.Errorf("none out of the seeds has a matching provider for %q", shoot.Spec.Provider.Type)
```

Comment (rfranzke) [reply pkg/scheduler/controller/shoot/reconciler.go:230]: This is in line with other similar statements, see

https://github.com/gardener/gardener/blob/36354063ffb7d173efab4db5c6999a291e8f25c0/pkg/scheduler/controller/shoot/reconciler.go#L189
https://github.com/gardener/gardener/blob/36354063ffb7d173efab4db5c6999a291e8f25c0/pkg/scheduler/controller/shoot/reconciler.go#L253
https://github.com/gardener/gardener/blob/36354063ffb7d173efab4db5c6999a291e8f25c0/pkg/scheduler/controller/shoot/reconciler.go#L259

Comment (acumino) [reply pkg/scheduler/controller/shoot/reconciler.go:230]: For consistency it's fine. But was there any reason to show no of seed?

**Review by acumino — COMMENTED** [review]
/lgtm

**Review by timebertt — APPROVED** [review]
/lgtm

**Pull request merged**